@utahdts/utah-design-system/react/components/forms/FileInput/FileInput.jsx

```diff
@@ -0,0 +1,171 @@
+ * @param {string} [props.className]
+ * @param {string} [props.errorMessage]
+ * @param {string} [props.hint]
+ * @param {string} props.id
+ * @param {import('react').Ref<HTMLDivElement>} [props.innerRef]
+ * @param {string} props.label
+ * @param {boolean} [props.multiple]
+ * @param {string} [props.name]
+ * @param {boolean} [props.isDisabled]
+ * @param {boolean} [props.isRequired]
+ * @param {import('react').ChangeEventHandler} [props.onChange]
+ * @param {FileList} [props.value]
+ * @returns {import('react').JSX.Element}
+ */
+export function FileInput({
+  acceptedFileTypes,
+  children,
+  className,
+  errorMessage,
+  hint,
+  id,
+  innerRef,
+  isDisabled,
+  isRequired,
+  label,
+  multiple,
+  name,
+  onChange,
+  value,
+}) {
+  const { addPoliteMessage } = useAriaMessaging();
+  const [isDragged, setDragged] = useImmer(false);
+  const [files, setFiles] = useImmer(value || []);
+  const inputRef = useRef(null);
+
+  const currentOnChange = useCallback(() => {
+    if (onChange) {
+      // @ts-ignore
+      onChange(inputRef.current.files);
+    } else {
+      // @ts-ignore
+      setFiles(inputRef.current.files);
+    }
+  }, []);
+
+  const removeFile = useCallback((/** @type {File} */ file) => {
+    // @ts-ignore
+    const currentFiles = [...inputRef.current.files];
+    const fileIndex = currentFiles.findIndex((item) => isEqual(file, item));
+    if (fileIndex !== -1) {
+      currentFiles.splice(fileIndex, 1);
+      // Create new FileList
+      const dataTransfer = new DataTransfer();
+      currentFiles.forEach((item) => dataTransfer.items.add(item));
+      // @ts-ignore
+      inputRef.current.files = dataTransfer.files;
+      currentOnChange();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (files.length) {
+      addPoliteMessage(`You have selected the file: ${[...files].map((item) => item.name).join(', ')}.`);
+    } else {
+      addPoliteMessage('No file selected.');
+    }
+  }, [files]);
+
+  useEffect(() => {
+    // Track files when controlled
+    if (onChange) { setFiles(value || []); }
+  }, [value, onChange]);
+
+  return (
+    <div className="input-wrapper" ref={innerRef}>
+      <label htmlFor={id}>
+        {label}
+        {isRequired ? <RequiredStar /> : null}
+      </label>
+      {hint
+        ? (
+          <div className="info-box file-input__info-box mt-spacing-xs">
+            <div className="info-box__content">
+              {hint}
+            </div>
+          </div>
+        )
+        : null}
+      <div className={joinClassNames(
+        'file-input__box mt-spacing-s',
+        isDragged ? 'file-input__box--dragged' : '',
+        isDisabled ? 'file-input__box--disabled' : ''
+      )}
+      >
+        <div className="file-input__safari" />
+        {!files.length
+          ? (
+            <div className="file-input__instructions">
+              <span className="button button--small">Drag file{multiple ? 's' : ''} here or click to upload</span>
```

Comment:
https://quillbot.com/grammar-check says that for singular it should read: 'Drag a file here'. Note the addition of the article 'a'.

Reply:
Yes.

Reply:
{multiple ? 'files' : 'a file'}